src/common/src/metrics.rs

```diff
@@ -44,3 +48,49 @@ impl Atomic for TrAdderAtomic {
 }
 
 pub type TrAdderGauge = GenericGauge<TrAdderAtomic>;
+
+pub struct CustomLayer {
+    pub aws_sdk_retry_counts: GenericCounter<AtomicU64>,
+    pub aws_http_timeout_retry_counts: GenericCounter<AtomicU64>,
+}
+
+impl<S> Layer<S> for CustomLayer
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+{
+    fn on_event(&self, event: &tracing::Event<'_>, _ctx: Context<'_, S>) {
+        if event.metadata().target() == "aws_smithy_client::retry"
+            && event.metadata().level() == &tracing::Level::DEBUG
+        {
+            self.aws_sdk_retry_counts.inc();
+        }
+
+        if event.metadata().target() == "http_timeout_retry"
```

Comment:
"http_timeout_retry" target is set by us in the s3 object store implementation, which is somehow too implicit. It is also implicit to capture tracing event in aws sdk but we currently don't have better way to do that so it is acceptable. But in here, I think we don't need to do that and subscribe the tracing event to emit this metric. We can just emit the metric in s3.rs.

Reply:
Make sense to me.